app/src/renderer/components/governance/PageProposal.vue

```diff
@@ -0,0 +1,196 @@
+  name: `page-proposal`,
+  props: [`proposal`],
+  components: {
+    TmBalance,
+    TmBtn,
+    FieldVote,
+    TmFormStruct,
+    TmToolBar,
+    TmPage,
+    TmPart,
+    TextBlock
+  },
+  computed: {
+    proposalType() {
+      return this.proposal.proposal_type.toLowerCase()
+    },
+    submitBlock() {
+      return `#` + num.prettyInt(this.proposal.submit_block)
+    },
+    voteBlock() {
+      if (this.proposal.submit_block === this.proposal.voting_start_block) {
+        return `the same block`
+      } else {
+        return `block #` + num.prettyInt(this.proposal.voting_start_block)
+      }
+    },
+    button() {
+      if (this.proposal.proposal_status === `Passed`) return null
+
+      if (this.proposal.proposal_status === `Rejected`) return null
+
+      if (this.proposal.proposal_status === `Active`) return `vote`
+
+      if (this.proposal.proposal_status === `Pending`) return `deposit`
+    },
+    status() {
+      if (this.proposal.proposal_status === `Passed`)
+        return `This proposal has passed`
+
+      if (this.proposal.proposal_status === `Rejected`)
+        return `This proposal has been rejected and voting is closed`
+
+      if (this.proposal.proposal_status === `Active`)
+        return `Voting for this proposal is open`
+
+      if (this.proposal.proposal_status === `Pending`)
+        return `Deposits are open for this proposal`
+    },
+    statusColor() {
+      if (this.proposal.proposal_status === `Passed`) return `green`
+      if (this.proposal.proposal_status === `Rejected`) return `red`
+      if (this.proposal.proposal_status === `Active`) return `blue`
+      if (this.proposal.proposal_status === `Pending`) return `yellow`
+    },
+    totalVotes() {
+      return (
+        this.proposal.tally_result.yes +
+        this.proposal.tally_result.no +
+        this.proposal.tally_result.no_with_veto +
+        this.proposal.tally_result.abstain
+      )
+    },
+    yesPercentage() {
+      return num.percentInt(this.proposal.tally_result.yes / this.totalVotes)
+    },
+    noPercentage() {
+      return num.percentInt(this.proposal.tally_result.no / this.totalVotes)
+    },
+    noWithVetoPercentage() {
+      return num.percentInt(
+        this.proposal.tally_result.no_with_veto / this.totalVotes
+      )
+    },
+    abstainPercentage() {
+      return num.percentInt(
+        this.proposal.tally_result.abstain / this.totalVotes
+      )
+    }
+  },
+  data: () => ({
+    votePick: `abstain`,
+    voteVisible: false
+  }),
+  methods: {
+    vote(choice) {
+      this.votePick = choice
+    },
+    confirmVote() {
+      this.$store.commit(`notify`, {
+        title: `Voted '${this.votePick}'`,
+        body: `Your vote has been submitted.`
+      })
+    },
+    commentOnProposal(proposalId) {
+      this.$store.commit(`notify`, {
+        title: `View Discussion Thread`,
+        body: `TODO: Discuss ${proposalId} proposal on the forum.`
+      })
+    },
+    proposalIsSpam(proposalId) {
```

Comment:
uhh I didn't know we where doing this. How would this work ? Comment on the forum ?